cocotb/runner.py

```diff
@@ -805,6 +804,112 @@ def test_command(self) -> List[Command]:
         return [[out_file] + self.plus_args]
 
 
+class Xcelium(Simulator):
+    @staticmethod
+    def simulator_in_path() -> None:
+        if shutil.which("xrun") is None:
+            raise SystemExit("ERROR: xrun executable not found!")
+
+    def check_toplevel_lang(self, toplevel_lang: Optional[str]) -> str:
+        if toplevel_lang is None:
+            if self.vhdl_sources and not self.verilog_sources:
+                return "vhdl"
+            elif self.verilog_sources and not self.vhdl_sources:
+                return "verilog"
+            else:
+                raise ValueError("Must specify a toplevel_lang in a mixed design")
+        elif toplevel_lang in ("verilog", "vhdl"):
+            return toplevel_lang
+        else:
+            raise ValueError(
+                f"Xcelium does not support {toplevel_lang!r} as a toplevel_lang"
+            )
+
+    @staticmethod
+    def get_include_options(includes: Sequence[str]) -> List[str]:
+        return [f"-incdir {include}" for include in includes]
+
+    @staticmethod
+    def get_define_options(defines: Sequence[str]) -> List[str]:
+        return [f"-define {define}" for define in defines]
+
+    @staticmethod
+    def get_parameter_options(parameters: Mapping[str, object]) -> List[str]:
+        return [f'-gpg "{name} => {value}"' for name, value in parameters.items()]
+
+    def build_command(self) -> List[Command]:
+
+        self.env["CDS_AUTO_64BIT"] = "all"
+        cmd = [
+            ["xrun"]
+            + ["-logfile xrun_build.log"]
+            + ["-elaborate"]
+            + [f"-xmlibdirname {self.build_dir}/xrun_snapshot"]
+            + ["-licqueue"]
+            # TODO: way to switch to these verbose messages?:
```

Comment:
Can add parameters for this specific simulator something like :
```python3
    def __init__(self, *argv, verbose=False, **kwargs):
        self.verbose = verbose
        super().__init__(*argv, **kwargs)
```

Reply:
I was thinking about whether this should also be controllable by an environment variable - either an existing/documented one (`COCOTB_LOG_LEVEL`) or an independent new one...

Reply:
Could also add parameters via pytest plugin.